### PR TITLE
Gh 33 context class

### PR DESCRIPTION
**Pull request by GCHQDeveloper404**

### CODE REVIEW

File: common/src/main/java/uk/gov/gchq/palisade/Context.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        return new ToStringBuilder(this)
+                .append("contents", contents)
+                .toString();
+    }
+
+    public Object get(final String key) {
+        return contents.get(key);
+    }
+
+    public Object put(final String key, final Object value) {
```
Comment (p013570) [inline common/src/main/java/uk/gov/gchq/palisade/Context.java]: Perhaps we should return Context here to allow users to do:

new Context().justification("abcdefg").put("key1", "value1).put("key2", value2");

Comment (GCHQDeveloper404) [reply common/src/main/java/uk/gov/gchq/palisade/Context.java]: Yep.

File: common/src/main/java/uk/gov/gchq/palisade/Context.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@JsonPropertyOrder(value = {"class", "contents"}, alphabetic = true)
```
Comment (ac74475) [inline common/src/main/java/uk/gov/gchq/palisade/Context.java]: Can we have some documentation about what this class is for

**Comment by ac74475** [comment]
Have you checked whether any of the documentation needs updating?

File: common/src/main/java/uk/gov/gchq/palisade/Context.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A structure to store specific information that is required for palisade interactions with resources.
```
Comment (ac74475) [inline common/src/main/java/uk/gov/gchq/palisade/Context.java]: I think we need to specify that it is query time information, or required information for a particular deployment which is not a core requirement for all deployments

**Pull request merged**